scalableminds/webknossos
# Fix rare crash when viewing shared annotation

If a user (who is not the owner) views a shared annotation, editing the layer settings will cause failed edit-annotation requests. After several retries (will take 5 minutes and new edits will reset the timer), the sagas will crash since the user is not allowed to edit annotation properties (only the tracings itself may be changed).

The PR fixes this by:
- ensuring that the failed saga won't crash the entire app
- avoiding trying to save if it's not allowed in the first place

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- create a shared annotation
- ensure that the dataset is accessible to the sample2 user
- log in as sample2
- ensure that the UI doesn't expose the edit annotation name and description buttons
- ensure that changing the layer visibility doesn't cause failed network requests

### Issues:
- fixes #6889

------
(Please delete unneeded items, merge only when none are left open)
